### PR TITLE
PYTHON-1079: Abstract Host Connection information

### PR DESCRIPTION
**Pull request by aboudreault**
* Add Host.host_id metadata
* Add Host.broadcast_rpc_address metadata
* Abstract Host connection information, now uses connection.EndPoint
* Add Cluster.endpoint_factory

### CODE REVIEW

File: cassandra/pool.py [hunk 1]
```diff
@@ -60,6 +60,11 @@ class Host(object):
     :attr:`~.Cluster.token_metadata_enabled` is ``False``.
     """
 
+    broadcast_rpc_address = None
+    """
+    The broadcast rpc address of the node (`native_address` or `rpc_address`).
+    """
+
```
Comment (aboudreault) [inline cassandra/pool.py]: This was basically `address` before, but untranslated.

File: cassandra/cluster.py [hunk 2]
```diff
@@ -2808,9 +2829,9 @@ class ControlConnection(object):
     """
 
     _SELECT_PEERS = "SELECT * FROM system.peers"
-    _SELECT_PEERS_NO_TOKENS = "SELECT peer, data_center, rack, rpc_address, release_version, schema_version FROM system.peers"
+    _SELECT_PEERS_NO_TOKENS = "SELECT host_id, peer, data_center, rack, rpc_address, release_version, schema_version FROM system.peers"
     _SELECT_LOCAL = "SELECT * FROM system.local WHERE key='local'"
-    _SELECT_LOCAL_NO_TOKENS = "SELECT cluster_name, data_center, rack, partitioner, release_version, schema_version FROM system.local WHERE key='local'"
+    _SELECT_LOCAL_NO_TOKENS = "SELECT host_id, rpc_address, cluster_name, data_center, rack, partitioner, release_version, schema_version FROM system.local WHERE key='local'"
```
Comment (aboudreault) [inline cassandra/cluster.py]: I think C* < 2.1.6 doesn't have this column.. need to find a solution

File: build.yaml [hunk 3]
```diff
@@ -43,6 +43,18 @@ schedules:
...
+    branches:
+      include: [/dev.*/]
+    env_vars: |
+      EVENT_LOOP_MANAGER='libev'
+      EXCLUDE_LONG=1
+    matrix:
+      exclude:
+        - python: [2.7, 3.4, 3.6]
+        - cassandra: ['2.0', '2.1', '2.2', '3.0', 'test-dse']
+
```
Comment (aboudreault) [inline build.yaml]: dev purpose... when I code.. I only one to see results for the latest versions.. not a whole matrix. will be rename to dev-python-*

**Comment by aboudreault** [comment]
Addressed comments. @mambocab  Can you take a quick look at my last commits?

**Comment by mambocab** [comment]
+1 on the new changes after a clean build!

**Pull request merged**